### PR TITLE
MULE-11735: Implement object streaming

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/util/FuncUtils.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.mule.runtime.core.util.func.CheckedRunnable;
+
+import java.util.function.Consumer;
+
+/**
+ * Utilities for functional programming
+ *
+ * @since 4.0
+ */
+public class FuncUtils {
```
Comment (pablolagreca) [inline core/src/main/java/org/mule/runtime/core/util/FuncUtils.java]: FunctionUtils

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/streaming/BytesStreamingExtensionTestCase.java [hunk 2]
```diff
@@ -107,6 +99,12 @@ public void seekInTx() throws Exception {
     doSeek("seekStreamTx");
   }
 
+  @Test
+  public void throwsBufferSizeExceededError() throws Exception {
```
Comment (pablolagreca) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/streaming/BytesStreamingExtensionTestCase.java]: Allure

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/streaming/ObjectStreamingExtensionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import org.mule.runtime.core.api.Event;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+public class ObjectStreamingExtensionTestCase extends AbstractStreamingExtensionTestCase {
```
Comment (pablolagreca) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/streaming/ObjectStreamingExtensionTestCase.java]: Allure

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/SpringMuleContextServiceConfigurator.java [hunk 4]
```diff
@@ -247,7 +248,7 @@ void createArtifactServices() {
 
   private void loadServiceConfigurators() {
     new SpiServiceRegistry()
-        .lookupProviders(ServiceConfigurator.class, Thread.currentThread().getContextClassLoader())
+        .lookupProviders(ServiceConfigurator.class, Service.class.getClassLoader())
```
Comment (pablolagreca) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/SpringMuleContextServiceConfigurator.java]: Is this intentional?

Comment (marianogonzalez) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/SpringMuleContextServiceConfigurator.java]: yes. If not EE doesn't pick up some configurators

**Pull request merged**